lib/pry/history.rb

```diff
@@ -97,6 +101,7 @@ def should_ignore?(line)
       hist_ignore = Pry.config.history.histignore
       return false if hist_ignore.nil? || hist_ignore.empty?
 
+      return true if should_add_duplicates? || ( @history.last && line == @history.last )
```

Comment:
I think this line should go before the one above it. Or the duplicate check will fail if there are no ignores.

Further, it should read like this:

```rb
return true if !should_add_duplicates? && (@history.last && line == @history.last)
```

Can `line` be falsy, BTW?